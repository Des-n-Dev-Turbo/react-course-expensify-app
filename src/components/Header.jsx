import React from "react";
import {
  NavLink
} from "react-router-dom";
import { connect } from 'react-redux';
import { startLogout } from "../actions/auth";

const Header = ({ startLogout }) => (
  <div>
    <h1>Extensify</h1>
    <NavLink to="/dashboard" exact activeClassName="is-active">
      Home
    </NavLink>
    <NavLink to="/create" activeClassName="is-active">
      Create
    </NavLink>
    <button onClick={startLogout}>Logout</button>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);
