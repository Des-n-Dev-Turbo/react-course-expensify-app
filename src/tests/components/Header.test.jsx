import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Header';
import toJSON from 'enzyme-to-json';

test('Should render Header Correctly', () => {
    const wrapper = shallow(<Header />);
    expect(toJSON(wrapper)).toMatchSnapshot();
});
