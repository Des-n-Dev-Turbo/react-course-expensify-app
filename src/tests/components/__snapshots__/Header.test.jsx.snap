// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header Correctly 1`] = `
<div>
  <h1>
    Extensify
  </h1>
  <NavLink
    activeClassName="is-active"
    exact={true}
    to="/"
  >
    Home
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/create"
  >
    Create
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/help"
  >
    Help
  </NavLink>
</div>
`;
